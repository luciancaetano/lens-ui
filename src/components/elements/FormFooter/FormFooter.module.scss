.form-footer {
  display: flex;
  color: var(--lens-ui-elements-divider-text-color);

  &__delete-container {
    display: flex;
    flex: 1;
  }

  &__button {
    margin-right: .625rem;
  }
}