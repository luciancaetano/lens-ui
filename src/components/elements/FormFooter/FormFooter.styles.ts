import styled from 'styled-components';

export const FormFooter = styled.div`
  display: flex;
  color: var(--lens-ui-elements-divider-text-color);

  .lens-ui-form-footer__delete-container {
    display: flex;
    flex: 1;
  }

  .lens-ui-form-footer-button {
    margin-right: .625rem;
  }
`;

export default { FormFooter };
