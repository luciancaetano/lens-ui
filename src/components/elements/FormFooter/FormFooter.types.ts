import React from 'react';
import { ITestableProps } from '../../../types';
import { ButtonAppearanceType } from '../Button/Button.types';

interface IFormFooterLocale {
  update: JSX.Element | JSX.Element[];
  updating: JSX.Element | JSX.Element[];
  save: JSX.Element | JSX.Element[];
  saving: JSX.Element | JSX.Element[];
  clear: JSX.Element | JSX.Element[];
  cleaning: JSX.Element | JSX.Element[];
  canceling: JSX.Element | JSX.Element[];
  cancel: JSX.Element | JSX.Element[];
  delete: JSX.Element | JSX.Element[];
  deleting: JSX.Element | JSX.Element[];
}

export interface IFormFooterProps extends ITestableProps, React.HtmlHTMLAttributes<HTMLElement> {
  saveAppearance?: ButtonAppearanceType;
  cancelAppearance?: ButtonAppearanceType;
  deleteAppearance?: ButtonAppearanceType;
  isUpdate?: boolean;
  state: 'saving' | 'deleting' | 'canceling' | 'ready';
  onSave?: (() => void) | 'submit';
  onCancel?: () => void;
  onDelete?: () => void;
  saveDisabled?: boolean;
  cancelDisabled?: boolean;
  deleteDisabled?: boolean;
  cancelIsReset?: boolean;
  childrenPos?: 'afterDelete' | 'beforeDelete' | 'afterCancel' | 'beforeCancel' | 'betweenSaveAndCancel' | 'afterSave';
  locale?: IFormFooterLocale;
}
