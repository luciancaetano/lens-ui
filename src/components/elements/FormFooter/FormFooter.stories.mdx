import { Canvas, Meta, Story } from '@storybook/addon-docs';
import ThemeProvider from '../ThemeProvider/ThemeProvider';
import FormFooter from './FormFooter';


<Meta title="Data Entry/FormFooter" component={FormFooter}/>


# FormFooter

FormFooter display's form actions like, save, cancel delete etc.

<Canvas>
    <FormFooter
      state="ready"
      onSave={() => {}}
      onCancel={() => {}}
      onDelete={() => {}}
    />
</Canvas>

### Properties

|Property|Type|Default Value|Description|Required|
|---|---|---|---|---|
|saveAppearance|`outlined`, `default`, `minimal`|'default'|Appearance of the save button|No|
|cancelAppearance|`outlined`, `default`, `minimal`|'default'|Appearance of the cancel button|No|
|deleteAppearance|`outlined`, `default`, `minimal`|'default'|Appearance of the delete button|No|
|isUpdate|boolean|false|If true, the delete button will be shown|No|
|state|`saving`, `deleting`, `canceling`, `ready`|'ready'|State of the form footer|No|
|onSave|function|undefined|Callback function to be called when the save button is clicked|No|
|onCancel|function|undefined|Callback function to be called when the cancel button is clicked|No|
|onDelete|function|undefined|Callback function to be called when the delete button is clicked|No|
|saveDisabled|boolean|false|If true, the save button will be disabled|No|
|cancelDisabled|boolean|false|If true, the cancel button will be disabled|No|
|deleteDisabled|boolean|false|If true, the delete button will be disabled|No|
|cancelIsReset|boolean|false|If true, the cancel button will be reset|No|
|childrenPos|`afterDelete`, `beforeDelete`, `afterCancel`, `beforeCancel`, `betweenSaveAndCancel`, `afterSave`|'afterSave'|Position of the children|No|
|locale|IFormFooterLocale|-|Localization object|No|
|testingID| string | undefined | The id used for testing purposes.<br/>`<div data-testid="my-test-id"/>` |No|

## IFormFooterLocale (Object)

### Properties

|Property|Type|Default Value|Description|Required|
|---|---|---|---|---|
|update|ReactNode|'Update'|Text for the update button|No|
|updating|ReactNode|'Updating'|Text for the updating button|No|
|save|ReactNode|'Save'|Text for the save button|No|
|saving|ReactNode|'Saving'|Text for the saving button|No|
|clear|ReactNode|'Clear'|Text for the clear button|No|
|cleaning|ReactNode|'Cleaning'|Text for the cleaning button|No|
|canceling|ReactNode|'Canceling'|Text for the canceling button|No|
|cancel|ReactNode|'Cancel'|Text for the cancel button|No|
|delete|ReactNode|'Delete'|Text for the delete button|No|
|deleting|ReactNode|'Deleting'|Text for the deleting button|No|

### Theme dark

<Canvas className="dark">
  <ThemeProvider theme="dark">
    <FormFooter
      state="ready"
      onSave={() => {}}
      onCancel={() => {}}
      onDelete={() => {}}
    />
  </ThemeProvider>
</Canvas>
