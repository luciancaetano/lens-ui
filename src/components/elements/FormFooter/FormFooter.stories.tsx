/* eslint-disable react/destructuring-assignment */
import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import FormFooter from './FormFooter';
import { IFormFooterProps } from './FormFooter.types';
import LensProvider from '../../providers/LensProvider/LensProvider';
import '../../../styles';

export default {
  title: '2. Components/FormFooter',
  component: FormFooter,
  decorators: [
    (Story) => <LensProvider>{Story()}</LensProvider>,
  ],
} as ComponentMeta<typeof FormFooter>;

const Template: ComponentStory<typeof FormFooter> = (args) => (
  <div style={{ maxWidth: 500 }}>
    <FormFooter {...args} />
  </div>
);

export const Default = Template.bind({});

Default.args = {
  onSave: () => {},
  onCancel: () => {},
  onDelete: () => {},
  state: 'ready',
} as IFormFooterProps;
