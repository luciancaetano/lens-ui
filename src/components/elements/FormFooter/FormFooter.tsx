import clsx from 'clsx';
import React, { useCallback, useMemo } from 'react';
import { useInternalLensLocale } from '../../..';
import Button from '../Button/Button';
import styles from './FormFooter.styles';
import { IFormFooterProps } from './FormFooter.types';
import { CLASSES } from '../../../css-classes';

const FormFooter:React.FC<IFormFooterProps> = ({
  className, testingID, id, onCancel, onDelete, onSave = 'submit', isUpdate, state,
  cancelAppearance, cancelDisabled, deleteAppearance, deleteDisabled,
  saveAppearance, saveDisabled, cancelIsReset,
  children, childrenPos = 'betweenSaveAndCancel',
}) => {
  const [_] = useInternalLensLocale();

  const saveButtonLabel = useMemo(() => {
    if (state === 'saving') {
      return isUpdate ? _('FormFooter/updating') : _('FormFooter/saving');
    } if (isUpdate) {
      return _('FormFooter/update');
    }

    return _('FormFooter/save');
  }, [state, _, isUpdate]);

  const cancelButtonLabel = useMemo(() => {
    if (state === 'canceling') return cancelIsReset ? _('FormFooter/cleaning') : _('FormFooter/cancel=>loading');
    return cancelIsReset ? _('FormFooter/clear') : _('FormFooter/cancel');
  }, [state, _, cancelIsReset]);

  const deleteButtonLabel = useMemo(() => {
    if (state === 'deleting') return _('FormFooter/deleting');
    return _('FormFooter/delete');
  }, [state, _]);

  const handleSave = useCallback(() => {
    if (typeof onSave === 'function') {
      onSave();
    }
  }, [onSave]);

  const handleCancel = useCallback(() => {
    if (typeof onCancel === 'function') {
      onCancel();
    }
  }, [onCancel]);

  const handleDelete = useCallback(() => {
    if (typeof onDelete === 'function') {
      onDelete();
    }
  }, [onDelete]);

  const cancelButtonType = useMemo(() => cancelIsReset ? 'reset' : 'button', [cancelIsReset]);

  const isLoading = useMemo(() => state !== 'ready', [state]);

  return (
    <styles.FormFooter id={id} data-testid={testingID} className={clsx(CLASSES.FontReset, 'lens-ui-form-footer', className)}>
      <div className="lens-ui-form-footer__delete-container">
        {childrenPos === 'beforeDelete' && children}
        {onDelete && (
          <Button
            className="lens-ui-form-footer-button lens-ui-form-footer-button__delete"
            intent="danger"
            appearance={deleteAppearance}
            disabled={deleteDisabled || isLoading}
            onClick={handleDelete}
          >{deleteButtonLabel}
          </Button>
        )}
        {childrenPos === 'afterDelete' && children}
      </div>
      {childrenPos === 'beforeCancel' && children}
      {onCancel && (
        <Button
          className="lens-ui-form-footer-button lens-ui-form-footer-button__cancel"
          intent="warning"
          type={cancelButtonType}
          appearance={cancelAppearance}
          disabled={cancelDisabled || isLoading}
          onClick={handleCancel}
        >{cancelButtonLabel}
        </Button>
      )}
      {childrenPos === 'betweenSaveAndCancel' && children}
      {onSave && (
        <Button
          className="lens-ui-form-footer-button lens-ui-form-footer-button__save"
          intent="success"
          type={onSave === 'submit' ? 'submit' : 'button'}
          appearance={saveAppearance}
          disabled={saveDisabled || isLoading}
          onClick={handleSave}
        >{saveButtonLabel}
        </Button>
      )}
      {childrenPos === 'afterSave' && children}
    </styles.FormFooter>
  );
};

export default FormFooter;
