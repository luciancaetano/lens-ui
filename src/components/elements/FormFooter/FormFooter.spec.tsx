import React from 'react';

import { render, fireEvent } from '@testing-library/react';
import FormFooter from './FormFooter';

describe('<FormFooter/>', () => {
  it('should check if the children content is present', () => {
    const childrenContent = 'My-Children-Component';
    const { getByText } = render(
      <FormFooter state="ready" testingID="testing-target">{childrenContent}</FormFooter>,
    );

    expect(getByText(childrenContent)).toBeInTheDocument();
  });

  it('should render children positions', () => {
    const childrenContent = 'My-Children-Component';

    expect(render(
      <FormFooter state="ready" childrenPos="afterCancel" testingID="testing-target">{childrenContent}</FormFooter>,
    ).container.firstChild).toMatchSnapshot();

    expect(render(
      <FormFooter state="ready" childrenPos="afterDelete" testingID="testing-target">{childrenContent}</FormFooter>,
    ).container.firstChild).toMatchSnapshot();

    expect(render(
      <FormFooter state="ready" childrenPos="afterSave" testingID="testing-target">{childrenContent}</FormFooter>,
    ).container.firstChild).toMatchSnapshot();

    expect(render(
      <FormFooter state="ready" childrenPos="beforeCancel" testingID="testing-target">{childrenContent}</FormFooter>,
    ).container.firstChild).toMatchSnapshot();

    expect(render(
      <FormFooter state="ready" childrenPos="beforeDelete" testingID="testing-target">{childrenContent}</FormFooter>,
    ).container.firstChild).toMatchSnapshot();

    expect(render(
      <FormFooter state="ready" childrenPos="betweenSaveAndCancel" testingID="testing-target">{childrenContent}</FormFooter>,
    ).container.firstChild).toMatchSnapshot();
  });

  it('should validate "ready" state', () => {
    const { container } = render(
      <FormFooter state="ready" testingID="testing-target" />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should validate "saving" state', () => {
    const { container } = render(
      <FormFooter state="saving" testingID="testing-target" />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should validate "saving" state when isUpdate', () => {
    const { container } = render(
      <FormFooter state="saving" isUpdate testingID="testing-target" />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should validate "ready" state when isUpdate', () => {
    const { container } = render(
      <FormFooter state="ready" isUpdate testingID="testing-target" />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should validate "saving" state when cancelIsReset is true', () => {
    const { container } = render(
      <FormFooter state="saving" cancelIsReset testingID="testing-target" />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should validate "saving" state when cancelIsReset is false', () => {
    const { container } = render(
      <FormFooter state="saving" cancelIsReset={false} testingID="testing-target" />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should validate "deleting" state', () => {
    const { container } = render(
      <FormFooter state="deleting" testingID="testing-target" />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should validate "canceling" state', () => {
    const { container } = render(
      <FormFooter state="canceling" testingID="testing-target" />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should validate "cancelIsReset" option', () => {
    const { container } = render(
      <FormFooter state="canceling" cancelIsReset testingID="testing-target" />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should receive buttons clicks', () => {
    const onSave = jest.fn();
    const onCancel = jest.fn();
    const onDelete = jest.fn();

    const { container } = render(
      <FormFooter
        state="ready"
        onSave={onSave}
        onCancel={onCancel}
        onDelete={onDelete}
        testingID="testing-target"
      />,
    );

    fireEvent.click(container.querySelector('[data-lens-element-button-parent="form-footer__save"]'));
    fireEvent.click(container.querySelector('[data-lens-element-button-parent="form-footer__cancel"]'));
    fireEvent.click(container.querySelector('[data-lens-element-button-parent="form-footer__delete"]'));

    expect(onSave).toHaveBeenCalledTimes(1);
    expect(onCancel).toHaveBeenCalledTimes(1);
    expect(onDelete).toHaveBeenCalledTimes(1);
  });

  it('should not receive buttons clicks when is state is saving', () => {
    const onSave = jest.fn();
    const onCancel = jest.fn();
    const onDelete = jest.fn();

    const result = render(
      <FormFooter
        state="saving"
        onSave={onSave}
        onCancel={onCancel}
        onDelete={onDelete}
        testingID="testing-target"
      />,
    );

    fireEvent.click(result.container.querySelector('[data-lens-element-button-parent="form-footer__save"]'));
    fireEvent.click(result.container.querySelector('[data-lens-element-button-parent="form-footer__cancel"]'));
    fireEvent.click(result.container.querySelector('[data-lens-element-button-parent="form-footer__delete"]'));

    expect(onSave).toHaveBeenCalledTimes(0);
    expect(onCancel).toHaveBeenCalledTimes(0);
    expect(onDelete).toHaveBeenCalledTimes(0);
  });

  it('should not receive buttons clicks when is state is deleting', () => {
    const onSave = jest.fn();
    const onCancel = jest.fn();
    const onDelete = jest.fn();

    const result = render(
      <FormFooter
        state="deleting"
        onSave={onSave}
        onCancel={onCancel}
        onDelete={onDelete}
        testingID="testing-target"
      />,
    );

    fireEvent.click(result.container.querySelector('[data-lens-element-button-parent="form-footer__save"]'));
    fireEvent.click(result.container.querySelector('[data-lens-element-button-parent="form-footer__cancel"]'));
    fireEvent.click(result.container.querySelector('[data-lens-element-button-parent="form-footer__delete"]'));

    expect(onSave).toHaveBeenCalledTimes(0);
    expect(onCancel).toHaveBeenCalledTimes(0);
    expect(onDelete).toHaveBeenCalledTimes(0);
  });

  it('should not receive buttons clicks when is state is canceling', () => {
    const onSave = jest.fn();
    const onCancel = jest.fn();
    const onDelete = jest.fn();

    const result = render(
      <FormFooter
        state="canceling"
        onSave={onSave}
        onCancel={onCancel}
        onDelete={onDelete}
        testingID="testing-target"
      />,
    );

    fireEvent.click(result.container.querySelector('[data-lens-element-button-parent="form-footer__save"]'));
    fireEvent.click(result.container.querySelector('[data-lens-element-button-parent="form-footer__cancel"]'));
    fireEvent.click(result.container.querySelector('[data-lens-element-button-parent="form-footer__delete"]'));

    expect(onSave).toHaveBeenCalledTimes(0);
    expect(onCancel).toHaveBeenCalledTimes(0);
    expect(onDelete).toHaveBeenCalledTimes(0);
  });
});
