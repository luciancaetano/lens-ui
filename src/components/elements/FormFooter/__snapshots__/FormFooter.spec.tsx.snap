// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormFooter/> should render children positions 1`] = `
<div
  class="formFooter"
  data-lens-element="form-footer"
  data-testid="testing-target"
>
  <div
    class="formFooterDeleteContainer"
  />
  <button
    class="button button--intent-success button--size-normal button--appearance-default button--intent-success--appearance-default formFooterButton"
    data-lens-appearance="default"
    data-lens-button-size="normal"
    data-lens-element="button"
    data-lens-element-button-parent="form-footer__save"
    data-lens-intent="success"
    type="submit"
  >
    Save
  </button>
</div>
`;

exports[`<FormFooter/> should render children positions 2`] = `
<div
  class="formFooter"
  data-lens-element="form-footer"
  data-testid="testing-target"
>
  <div
    class="formFooterDeleteContainer"
  >
    My-Children-Component
  </div>
  <button
    class="button button--intent-success button--size-normal button--appearance-default button--intent-success--appearance-default formFooterButton"
    data-lens-appearance="default"
    data-lens-button-size="normal"
    data-lens-element="button"
    data-lens-element-button-parent="form-footer__save"
    data-lens-intent="success"
    type="submit"
  >
    Save
  </button>
</div>
`;

exports[`<FormFooter/> should render children positions 3`] = `
<div
  class="formFooter"
  data-lens-element="form-footer"
  data-testid="testing-target"
>
  <div
    class="formFooterDeleteContainer"
  />
  <button
    class="button button--intent-success button--size-normal button--appearance-default button--intent-success--appearance-default formFooterButton"
    data-lens-appearance="default"
    data-lens-button-size="normal"
    data-lens-element="button"
    data-lens-element-button-parent="form-footer__save"
    data-lens-intent="success"
    type="submit"
  >
    Save
  </button>
  My-Children-Component
</div>
`;

exports[`<FormFooter/> should render children positions 4`] = `
<div
  class="formFooter"
  data-lens-element="form-footer"
  data-testid="testing-target"
>
  <div
    class="formFooterDeleteContainer"
  />
  My-Children-Component
  <button
    class="button button--intent-success button--size-normal button--appearance-default button--intent-success--appearance-default formFooterButton"
    data-lens-appearance="default"
    data-lens-button-size="normal"
    data-lens-element="button"
    data-lens-element-button-parent="form-footer__save"
    data-lens-intent="success"
    type="submit"
  >
    Save
  </button>
</div>
`;

exports[`<FormFooter/> should render children positions 5`] = `
<div
  class="formFooter"
  data-lens-element="form-footer"
  data-testid="testing-target"
>
  <div
    class="formFooterDeleteContainer"
  >
    My-Children-Component
  </div>
  <button
    class="button button--intent-success button--size-normal button--appearance-default button--intent-success--appearance-default formFooterButton"
    data-lens-appearance="default"
    data-lens-button-size="normal"
    data-lens-element="button"
    data-lens-element-button-parent="form-footer__save"
    data-lens-intent="success"
    type="submit"
  >
    Save
  </button>
</div>
`;

exports[`<FormFooter/> should render children positions 6`] = `
<div
  class="formFooter"
  data-lens-element="form-footer"
  data-testid="testing-target"
>
  <div
    class="formFooterDeleteContainer"
  />
  My-Children-Component
  <button
    class="button button--intent-success button--size-normal button--appearance-default button--intent-success--appearance-default formFooterButton"
    data-lens-appearance="default"
    data-lens-button-size="normal"
    data-lens-element="button"
    data-lens-element-button-parent="form-footer__save"
    data-lens-intent="success"
    type="submit"
  >
    Save
  </button>
</div>
`;

exports[`<FormFooter/> should validate "cancelIsReset" option 1`] = `
<div
  class="formFooter"
  data-lens-element="form-footer"
  data-testid="testing-target"
>
  <div
    class="formFooterDeleteContainer"
  />
  <button
    class="button button--intent-success button--size-normal button--appearance-default button--intent-success--appearance-default formFooterButton"
    data-lens-appearance="default"
    data-lens-button-size="normal"
    data-lens-element="button"
    data-lens-element-button-parent="form-footer__save"
    data-lens-intent="success"
    disabled=""
    type="submit"
  >
    Save
  </button>
</div>
`;

exports[`<FormFooter/> should validate "canceling" state 1`] = `
<div
  class="formFooter"
  data-lens-element="form-footer"
  data-testid="testing-target"
>
  <div
    class="formFooterDeleteContainer"
  />
  <button
    class="button button--intent-success button--size-normal button--appearance-default button--intent-success--appearance-default formFooterButton"
    data-lens-appearance="default"
    data-lens-button-size="normal"
    data-lens-element="button"
    data-lens-element-button-parent="form-footer__save"
    data-lens-intent="success"
    disabled=""
    type="submit"
  >
    Save
  </button>
</div>
`;

exports[`<FormFooter/> should validate "deleting" state 1`] = `
<div
  class="formFooter"
  data-lens-element="form-footer"
  data-testid="testing-target"
>
  <div
    class="formFooterDeleteContainer"
  />
  <button
    class="button button--intent-success button--size-normal button--appearance-default button--intent-success--appearance-default formFooterButton"
    data-lens-appearance="default"
    data-lens-button-size="normal"
    data-lens-element="button"
    data-lens-element-button-parent="form-footer__save"
    data-lens-intent="success"
    disabled=""
    type="submit"
  >
    Save
  </button>
</div>
`;

exports[`<FormFooter/> should validate "ready" state 1`] = `
<div
  class="formFooter"
  data-lens-element="form-footer"
  data-testid="testing-target"
>
  <div
    class="formFooterDeleteContainer"
  />
  <button
    class="button button--intent-success button--size-normal button--appearance-default button--intent-success--appearance-default formFooterButton"
    data-lens-appearance="default"
    data-lens-button-size="normal"
    data-lens-element="button"
    data-lens-element-button-parent="form-footer__save"
    data-lens-intent="success"
    type="submit"
  >
    Save
  </button>
</div>
`;

exports[`<FormFooter/> should validate "ready" state when isUpdate 1`] = `
<div
  class="formFooter"
  data-lens-element="form-footer"
  data-testid="testing-target"
>
  <div
    class="formFooterDeleteContainer"
  />
  <button
    class="button button--intent-success button--size-normal button--appearance-default button--intent-success--appearance-default formFooterButton"
    data-lens-appearance="default"
    data-lens-button-size="normal"
    data-lens-element="button"
    data-lens-element-button-parent="form-footer__save"
    data-lens-intent="success"
    type="submit"
  >
    Update
  </button>
</div>
`;

exports[`<FormFooter/> should validate "saving" state 1`] = `
<div
  class="formFooter"
  data-lens-element="form-footer"
  data-testid="testing-target"
>
  <div
    class="formFooterDeleteContainer"
  />
  <button
    class="button button--intent-success button--size-normal button--appearance-default button--intent-success--appearance-default formFooterButton"
    data-lens-appearance="default"
    data-lens-button-size="normal"
    data-lens-element="button"
    data-lens-element-button-parent="form-footer__save"
    data-lens-intent="success"
    disabled=""
    type="submit"
  >
    Saving
  </button>
</div>
`;

exports[`<FormFooter/> should validate "saving" state when cancelIsReset is false 1`] = `
<div
  class="formFooter"
  data-lens-element="form-footer"
  data-testid="testing-target"
>
  <div
    class="formFooterDeleteContainer"
  />
  <button
    class="button button--intent-success button--size-normal button--appearance-default button--intent-success--appearance-default formFooterButton"
    data-lens-appearance="default"
    data-lens-button-size="normal"
    data-lens-element="button"
    data-lens-element-button-parent="form-footer__save"
    data-lens-intent="success"
    disabled=""
    type="submit"
  >
    Saving
  </button>
</div>
`;

exports[`<FormFooter/> should validate "saving" state when cancelIsReset is true 1`] = `
<div
  class="formFooter"
  data-lens-element="form-footer"
  data-testid="testing-target"
>
  <div
    class="formFooterDeleteContainer"
  />
  <button
    class="button button--intent-success button--size-normal button--appearance-default button--intent-success--appearance-default formFooterButton"
    data-lens-appearance="default"
    data-lens-button-size="normal"
    data-lens-element="button"
    data-lens-element-button-parent="form-footer__save"
    data-lens-intent="success"
    disabled=""
    type="submit"
  >
    Saving
  </button>
</div>
`;

exports[`<FormFooter/> should validate "saving" state when isUpdate 1`] = `
<div
  class="formFooter"
  data-lens-element="form-footer"
  data-testid="testing-target"
>
  <div
    class="formFooterDeleteContainer"
  />
  <button
    class="button button--intent-success button--size-normal button--appearance-default button--intent-success--appearance-default formFooterButton"
    data-lens-appearance="default"
    data-lens-button-size="normal"
    data-lens-element="button"
    data-lens-element-button-parent="form-footer__save"
    data-lens-intent="success"
    disabled=""
    type="submit"
  >
    Updating
  </button>
</div>
`;
